cdap-proto/src/main/java/co/cask/cdap/proto/id/ProgramId.java

```diff
@@ -39,7 +39,7 @@ public ProgramId(String namespace, String application, ProgramType type, String
     this(new ApplicationId(namespace, application), type, program);
   }
 
-  ProgramId(ApplicationId appId, ProgramType type, String program) {
+  public ProgramId(ApplicationId appId, ProgramType type, String program) {
```

Comment:
I believe this was made package private on purpose. Instead of making it public, you can use appId.program(programType, programName).